pontis-package/pontis/core/const.py

```diff
@@ -7,3 +9,11 @@
 )
 NETWORK = "testnet"
 DEFAULT_AGGREGATION_MODE = 0
+
+if os.environ.get("__PONTIS_STAGING_ENV__") == "TRUE":
```

Comment:
This is the part where how you wrote the staging made sense to me. My initial intuition was to create a seperate directory adjacent to pontis core called staging which extended all the core contracts exactly and worked as a control space pointing to a diff addr. What it seems you did is just based on defined environment variables or mode, changed the pointed to addr, with the in line logic staying the same. 

I like your approach better (more concise) and I think you don't run into any problems with writing to cache variables since deployed Cairo acts as immutable ROM. That said, in cases where the staging execution path cuts across non-Cairo python scripts, is there any way that modifications to that runtime (maybe by different inputs or array/entry modification) could affect cached variables on the lightsail server? I don't know enough to be certain but if you are certain this is not an issue then feel free to merge.

Reply:
This is a clear response, thanks. Tangential question, would the best way to describe the "pontis-package/pontis/core/client.py" file (which the staging env changes the ```PUBLISHER_ADDRESS``` reference to) be as a python pseudo-frontend for calling the Cairo code? Or is client.py better described a python implementation of the proxy contract you define in the README.md schema/diagram (I don't think this is the case; implementation selection is different from changing between prod and stagenet, and the former is AFAIK managed by the Oracle Controller). 

Referring back to your response: I was asking about accidentally writing to prod when a publisher thinks they're on stagenet, though you're really covered on accidental stagenet writing.* What do you think about adding a command line echo to reaffirm whether you're in prod or stagenet, so as to prevent crossover in any case, if user forgets to clear ENV vars. 

*Explicitly checking for "TRUE" to exclude truthy values was a nice choice, so as to prevent inadvertent stage publishing :)

Reply:
Yes, the `PontisClient` is a "pseudo-frontend" (would usually call this an SDK) in the sense that its primary purpose is to package functionality involved in talking to our smart contracts to receive price data. It is not an implementation of on-chain logic, but merely a means to communicate with the contracts.

Good point, added in https://github.com/42labs/Pontis/pull/47/commits/89d610db8b6da0638830457a7c1cc95862f59d7c